src/adapters/WstEthChainlinkAdapter.sol

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity 0.8.21;
+
+import {IStEth} from "../interfaces/IStEth.sol";
+import {AggregatorV3Interface} from "../interfaces/AggregatorV3Interface.sol";
+
+import {ErrorsLib} from "../libraries/ErrorsLib.sol";
+
+/// @title WstEthChainlinkAdapter
+/// @author Morpho Labs
+/// @custom:contact security@morpho.org
+/// @notice wstETH/ETH exchange rate price feed.
+/// @dev This contract should only be used as price feed for `ChainlinkOracle`.
+contract WstEthChainlinkAdapter is AggregatorV3Interface {
+    uint8 public constant decimals = uint8(18);
+    string public constant description = "wstETH/ETH exchange rate";
+    uint256 public constant version = 1;
```

Comment:
The version number is supposed to be `representing the type of aggregator the proxy points to`
Because this is not targetting a particular Chainlink proxy, I think it would be more error proof to revert when looking at the version number

Reply:
> I think it would be more error proof to revert when looking at the version number

I guess you wanted to say the opposite no?

Reply:
Error proof, not error prone, so it would lead to less errors if it reverted instead

Reply:
oh thx misread it

Reply:
I agree w @QGarchery because this adapter is not an aggregator proxy, it should be safer to revert instead of displaying a random, inaccurate version number

Reply:
I'm surprised by this. Isn't there a chance to create errors by reverting? ie for any library that queries the version number, gets an error, and decides the contract is not spec compliant?

Reply:
Yes this is supposed to be for internal use only (ChainlinkOracle). And reverting makes sure that it does not get confused for a "real" chainlink feed